### PR TITLE
GSdx:Windows: Show different adapters for each renderer

### PR DESCRIPTION
**Pull request by turtleli**
Main change:
 * Show different adapters for each renderer. An adapter cannot be selected on OpenGL, and the available adapters available for D3D11 and D3D9 may be slightly different.

Other changes (that make the above change easier):
 * Don't refer to D3D10 - D3D11 is being used even if the user's GPU only supports D3D feature level 10.0.
 * Move D3D11 and D3D9 adapter enumeration code into separate functions.
 * Open sub dialogs using temporary objects - it simplifies the logic slightly as the dialog state won't be partially preserved if it's closed and reopened.
 * Initialise hack dialog variables in constructor - not exactly part of the PR, but it does kinda follow on from the previous change.

adapt

### CODE REVIEW

**Comment by lightningterror** [comment]
Maybe also grey out the _Adapter:_ text in gl ? Since that's how the rest of the gsdx options work.

File: plugins/GSdx/GSdx.rc [hunk 1]
```diff
@@ -286,14 +284,13 @@ FONT 8, "MS Shell Dlg", 400, 0, 0x1
...
     CONTROL         IDB_LOGO9,IDC_LOGO9,"Static",SS_BITMAP | SS_REALSIZECONTROL,6,6,230,44
-    CONTROL         IDB_LOGO10,IDC_LOGO10,"Static",SS_BITMAP | SS_REALSIZECONTROL,6,6,230,44
     CONTROL         IDB_LOGO11,IDC_LOGO11,"Static",SS_BITMAP | SS_REALSIZECONTROL,6,6,230,44
     CONTROL         IDB_LOGOGL,IDC_LOGOGL,"Static",SS_BITMAP | SS_REALSIZECONTROL,6,6,230,44
     CONTROL         IDB_NULL,IDC_NULL,"Static",SS_BITMAP | SS_REALSIZECONTROL,6,6,230,44
-    LTEXT           "Adapter:",IDC_STATIC,6,57,30,8
-    COMBOBOX        IDC_ADAPTER,71,55,166,118,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
-    LTEXT           "Renderer:",IDC_STATIC,6,72,34,8
-    COMBOBOX        IDC_RENDERER,71,70,166,118,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
+    LTEXT           "Renderer:",IDC_STATIC,6,57,30,8
```
Comment (turtleli) [inline plugins/GSdx/GSdx.rc]: Label length needs to be increased - `:` is not visible in the images.

**Comment by turtleli** [comment]
Updated to disable `Adapter:` text when OpenGL or Null is selected and to fix the label truncation issue.

**Comment by lightningterror** [comment]
While you're at it you could also disable/grey out Texture Filtering if you like.

**Comment by turtleli** [comment]
No thanks. ;)

I usually prefer to leave GSdx Windows GUI changes to other people.

**Comment by turtleli** [comment]
> What's the status ?

Not sure. Might make your PC implode.

**Comment by lightningterror** [comment]
> Not sure. Might make your PC implode.

http://www.nooooooooooooooo.com/

**Pull request merged**